### PR TITLE
BaseAPIClient: make writeable attributes private

### PR DESCRIPTION
**Pull request by risicle**
God knows where the ticket for this is if it ever had one - it's been kicking around for a while.

This is to encourage immutabile use of api clients after construction/init_app, which should provide more threadsafety.

### CODE REVIEW

File: dmapiclient/base.py [hunk 1]
```diff
@@ -49,17 +49,58 @@ def iter_method(self, *args, **kwargs):
...
+    @property
+    def enabled(self):
+        return self._enabled
+
+    @property
+    def timeout(self):
+        return self._timeout
+
+
+    __slots__ = ("_base_url", "_auth_token", "_enabled", "_timeout",)
```
Comment (katstevens) [inline dmapiclient/base.py]: This might be a dumb question: what is this used for?

Comment (risicle) [reply dmapiclient/base.py]: Giving a python class `__slots__` tells python not to give it a dictionary for implementing its attributes, instead giving it a fixed selection of attributes. Basically, "don't let people assign attributes on an instance other than these listed"

Comment (katstevens) [reply dmapiclient/base.py]: Aha, thanks! Every day is a school day.

Comment (risicle) [reply dmapiclient/base.py]: hmm unfortunately it seems to make `mock.patch.object` very unhappy.

**Pull request merged**